File: backend/app/Models/QuizSetting.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use App\Enums\VehicleType;
use App\Enums\QuizMode;

class QuizSetting extends Model
{
    use HasFactory;

    public $timestamps = false;

    protected $table = 'quiz_settings';

    protected $fillable = [
        'quiz_id',
        'vehicle_type',
        'question_count',
        'mode',
        'randomize_questions',
        'randomize_choices',
    ];

    protected $casts = [
        'vehicle_type' => VehicleType::class,
        'mode' => QuizMode::class,
        'randomize_questions' => 'boolean',
        'randomize_choices' => 'boolean',
        'question_count' => 'integer',
    ];

    public function quiz(): BelongsTo
    {
        return $this->belongsTo(Quiz::class);
    }
}

File: backend/app/models/quiz_setting.py
from __future__ import annotations

from typing import TYPE_CHECKING

from sqlalchemy import Boolean, Enum, Integer, text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base, BigIntPrimaryKeyMixin, SoftDeleteMixin, TimestampMixin
from app.models.enums import QuizMode, VehicleType, enum_values

if TYPE_CHECKING:
    from app.models.quiz import Quiz


class QuizSetting(BigIntPrimaryKeyMixin, TimestampMixin, SoftDeleteMixin, Base):
    __tablename__ = "quiz_setting"

    vehicle_type: Mapped[VehicleType] = mapped_column(
        Enum(VehicleType, name="quiz_settings_vehicle_type_enum", values_callable=enum_values),
        nullable=False,
        server_default=text(f"'{VehicleType.CAR.value}'"),
    )
    mode: Mapped[QuizMode] = mapped_column(
        Enum(QuizMode, name="quiz_settings_mode_enum", values_callable=enum_values),
        nullable=False,
        server_default=text(f"'{QuizMode.TRAINING.value}'"),
    )
    question_count: Mapped[int] = mapped_column(Integer, nullable=False, server_default=text("10"))
    randomize_questions: Mapped[bool] = mapped_column(Boolean, nullable=False, server_default=text("true"))
    randomize_choices: Mapped[bool] = mapped_column(Boolean, nullable=False, server_default=text("true"))
    passing_score: Mapped[int] = mapped_column(Integer, nullable=False, server_default=text("70"))
    review_allowed: Mapped[bool] = mapped_column(Boolean, nullable=False, server_default=text("false"))

    quiz: Mapped["Quiz | None"] = relationship(back_populates="setting", uselist=False)
